### PR TITLE
GRPC stub for the ResourceService

### PR DESCRIPTION
**Pull request by analogue**
### Description

Create a bare bones implementation of the `ResourceService` and wire up to the external GRPC server.

### Testing & Reproduction steps
- no-op test methods for each public function on `ResourceService` created

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: agent/consul/server.go [hunk 1]
```diff
@@ -1219,6 +1159,76 @@ func (s *Server) setupRPC() error {
...
+			// Only forward the request if the dc in the request matches the server's datacenter.
+			if info.RequestDatacenter() != "" && info.RequestDatacenter() != config.Datacenter {
+				return false, fmt.Errorf("requests to generate peering tokens cannot be forwarded to remote datacenters")
+			}
+			return s.ForwardGRPC(s.grpcConnPool, info, fn)
+		},
+	})
+	s.peerStreamServer.Register(s.externalGRPCServer)
+
+	resource.NewServer(resource.Config{}).Register(s.externalGRPCServer)
```
Comment (analogue) [inline agent/consul/server.go:1229]: This is the only new addition here outside of extracting code to `setupExternalGRPC(...)`

File: agent/grpc-middleware/rate_limit_mappings.gen.go [hunk 2]
```diff
@@ -22,6 +22,12 @@ var rpcRateLimitSpecs = map[string]rate.OperationType{
 	"/hashicorp.consul.internal.peering.PeeringService/TrustBundleRead":          rate.OperationTypeRead,
 	"/hashicorp.consul.internal.peerstream.PeerStreamService/ExchangeSecret":     rate.OperationTypeWrite,
 	"/hashicorp.consul.internal.peerstream.PeerStreamService/StreamResources":    rate.OperationTypeRead,
+	"/hashicorp.consul.resource.ResourceService/Delete":                          rate.OperationTypeWrite,
+	"/hashicorp.consul.resource.ResourceService/List":                            rate.OperationTypeRead,
+	"/hashicorp.consul.resource.ResourceService/Read":                            rate.OperationTypeRead,
+	"/hashicorp.consul.resource.ResourceService/Watch":                           rate.OperationTypeWrite,
```
Comment (boxofrad) [inline agent/grpc-middleware/rate_limit_mappings.gen.go:28]: Not for now, but at some point we should probably revisit how we apply rate-limits to the `Watch` endpoint, as it's really `len(event)` reads.

**Review by boxofrad — APPROVED** [review]
Awesome, thanks @analogue 🙌🏻

File: agent/grpc-middleware/rate_limit_mappings.gen.go (hunk 2 above)
Comment (analogue) [reply agent/grpc-middleware/rate_limit_mappings.gen.go]: Updated `Watch` to a read in https://github.com/hashicorp/consul/pull/16528/commits/73fee3ff8b6779124211ebfaa449c27afb3f1095

**Pull request merged**